### PR TITLE
kconfig: sca: Add Kconfig for SCA tools

### PR DESCRIPTION
**Pull request by simhein**
Add the possibility to use Kconfig for SCA tools.

Sometimes SCA tools can have a different or complex setups to run or outputs to generate.
The implemented cmake sca mechanism right now only allows configuration of the tools by using env variables or passing parameters to the build command. Which makes the integration and configuration seems very hidden and undocumented for everyone else as well. 

The first Kconfig options for a SCA tool would be introduced with https://github.com/zephyrproject-rtos/zephyr/pull/68324

### CODE REVIEW

File: cmake/sca/Kconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Static Code Analysis (SCA) options
+
+# Copyright (c) 2023 Baumer (www.baumer.com)
```
Comment (pdgendt) [inline cmake/sca/Kconfig:3]:
```suggestion
# Copyright (c) 2024 Baumer (www.baumer.com)
```

Comment (simhein) [reply cmake/sca/Kconfig]: fixed. Seems like I still living in 2023 😄

**Comment by pdgendt** [comment]
Adding this without any contents doesn't seem to be the right approach though. You should at least add one of the SCA tools to begin with IMO.

**Comment by simhein** [comment]
> Adding this without any contents doesn't seem to be the right approach though. You should at least add one of the SCA tools to begin with IMO.

Good point, thought about the same thing after I opened it yesterday.
I will close this PR and will add it with my PR https://github.com/zephyrproject-rtos/zephyr/pull/68324 where the first tool would use this

**Pull request closed without merging**